modules/vshn-lbaas-exoscale/hiera.tf

```diff
@@ -7,6 +7,49 @@ locals {
   ) : ""
   nat_vip = var.cluster_network.enabled ? exoscale_elastic_ip.nat[0].ip_address : ""
 }
+
+resource "exoscale_iam_role" "floaty" {
+  name        = "${var.cluster_id}_floaty"
+  description = "Exoscale IAMv3 role for Floaty for ${var.cluster_id}"
+  // TBD if we want to set `editable=false` -- note that this also prevents
+  // updates via Terraform
+  editable = true
+
+  policy = {
+    default_service_strategy = "deny"
+
+    services = {
+      compute-legacy = {
+        type = "rules"
+        rules = [
+          {
+            action     = "allow"
+            expression = "operation in ['compute-add-ip-to-nic', 'compute-list-nics', 'compute-list-resource-details', 'compute-list-virtual-machines', 'compute-query-async-job-result', 'compute-remove-ip-from-nic']"
+          }
+        ]
+      }
+      compute = {
+        type = "rules"
+        rules = [
+          {
+            action     = "allow"
+            expression = "operation in ['get-instance', 'list-instances', 'list-elastic-ips']"
+          },
+          {
+            action     = "allow"
+            expression = "operation in ['attach-instance-to-elastic-ip', 'detach-instance-from-elastic-ip'] && resources.elastic_ip.ip in ['${exoscale_elastic_ip.api.ip_address}', '${exoscale_elastic_ip.ingress.ip_address}']"
```

Comment:
Note that we allow Floaty to attach/detach the API and Ingress VIP to any instance that's accessible with the credentials. We currently can't sensibly restrict this to the LBs, since we need the instance IDs in the policy, but we want to create the credentials (and the Hieradata) before we create the instances so bootstrapping doesn't depend on a race condition.